mempool/reaper.go

```diff
@@ -0,0 +1,110 @@
+package mempool
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	cmtypes "github.com/cometbft/cometbft/types"
+
+	"github.com/cometbft/cometbft/libs/log"
+	seqGRPC "github.com/rollkit/go-sequencing/proxy/grpc"
+)
+
+// ReapInterval is the interval at which the reaper checks the mempool for transactions to reap.
+const (
+	ReapInterval time.Duration = 1 * time.Second
+	MaxRetries   int           = 3
+	RetryDelay   time.Duration = 2 * time.Second
+)
+
+// CListMempoolReaper is a reaper that reaps transactions from the mempool and sends them to the gRPC server.
+type CListMempoolReaper struct {
+	mempool    Mempool
+	stopCh     chan struct{}
+	grpcClient *seqGRPC.Client
+	ctx        context.Context
+	rollupId   []byte
+	submitted  map[cmtypes.TxKey]struct{}
+	mu         sync.RWMutex // Add a mutex to protect the submitted map
+	logger     log.Logger
+}
+
+// NewCListMempool initializes the mempool and sets up the gRPC client.
+func NewCListMempoolReaper(mempool Mempool, rollupId []byte, seqClient *seqGRPC.Client, logger log.Logger) *CListMempoolReaper {
+	return &CListMempoolReaper{
+		mempool:    mempool,
+		stopCh:     make(chan struct{}),
+		grpcClient: seqClient,
+		ctx:        context.Background(),
+		rollupId:   rollupId,
+		submitted:  make(map[cmtypes.TxKey]struct{}),
+		logger:     logger,
+	}
+}
+
+// StartReaper starts the reaper goroutine.
+func (r *CListMempoolReaper) StartReaper() error {
+	go func() {
+		ticker := time.NewTicker(ReapInterval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				r.reap()
+			case <-r.stopCh:
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+func (r *CListMempoolReaper) UpdateCommitedTxs(txs []cmtypes.Tx) {
+	r.mu.Lock() // Lock the mutex before modifying the map
+	defer r.mu.Unlock()
+	for _, tx := range txs {
+		delete(r.submitted, tx.Key())
+	}
+}
+
+// StopReaper stops the reaper goroutine.
+func (r *CListMempoolReaper) StopReaper() {
+	close(r.stopCh)
+}
```

Comment:
To address @tzdybal comment [here](https://github.com/rollkit/rollkit/pull/1789/files#r1737081977).
We should make the context a context with cancel and then cancel it here. 
Otherwise we are just passing a context that isn't stopping with the shutdown here reliably. 

I think if we don't cancel the context the context would hang around until the GC cleans it up. 
But the GC only cleans something when it is no longer being referenced, so as long as the called client is using it, it wouldn't be cleaned up. Which then makes it useless.